File: src/main/java/com/example/compas/util/AlternativePair.java
package com.example.compas.util;

import com.example.compas.entity.AlternativeEntity;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class AlternativePair {
    private AlternativeEntity first;
    private AlternativeEntity second;
}

File: shared/src/commonMain/kotlin/AlternativePair.kt
package org.example

import kotlinx.serialization.Serializable

@Serializable
data class AlternativePair(
    var first: AlternativeEntity,
    var second: AlternativeEntity
) {
    fun isEqual(other: Any?): Boolean {
        if (other == null || other !is AlternativePair) return false
        if (first.isEqual(other.first) && second.isEqual(other.second)) return true
        return first.isEqual(other.second) && second.isEqual(other.first)
    }
}
